src/store.rs

```diff
@@ -1169,633 +941,408 @@ impl<F: LurkField> Store<F> {
         }
     }
 
-    /// Mutable version of car_cdr to handle Str. `(cdr str)` may return a new str (the tail), which must be allocated.
-    pub fn car_cdr_mut(&mut self, ptr: &Ptr<F>) -> Result<(Ptr<F>, Ptr<F>), Error> {
-        match ptr.tag {
-            ExprTag::Nil => Ok((self.get_nil(), self.get_nil())),
-            ExprTag::Cons => match self.fetch(ptr) {
-                Some(Expression::Cons(car, cdr)) => Ok((car, cdr)),
-                Some(Expression::Opaque(_)) => Err(Error("cannot destructure opaque Cons".into())),
-                _ => unreachable!(),
-            },
-            ExprTag::Str => {
-                if let Some(Expression::Str(s)) = self.fetch(ptr) {
-                    let mut str = s.chars();
-                    if let Some(c) = str.next() {
-                        let cdr_str: String = str.collect();
-                        let str = self.intern_str(&cdr_str);
-                        Ok((self.get_char(c), str))
-                    } else {
-                        Ok((self.nil(), self.intern_str("")))
-                    }
-                } else {
-                    panic!();
+    // TODO: add cycle detection to avoid infinite recursion
```

Comment:
Could this be as simple as calling into a 
```rust
fn get_z_expr_aux(
        &self,
        ptr: &Ptr<F>,
        z_store: Option<Rc<RefCell<ZStore<F>>>>,
        visited: &mut HashSet<Ptr<F>>,
    ) -> Result<(ZExprPtr<F>, Option<ZExpr<F>>), Error> {
    ...
}
``` 
? Do we have an issue to track this?

Reply:
#479